src/components/pagination/nz-pagination.component.ts

```diff
@@ -192,10 +192,10 @@ export class NzPaginationComponent {
     if (this._current === value) {
       return;
     }
-    if (value > this._lastIndex || value < this._firstIndex) {
-      return;
+    if (value > this._lastIndex) {
+      this._current = this._lastIndex;
     }
-    this._current = Number(value);
+    this._current = value;
```

Comment:
This logic make `this._current` always be `value`

Reply:
According to the source logic, here is judging to avoid out of bounds

Reply:
I have rolled back.